app/models/my_academics/semesters.rb

```diff
@@ -153,22 +153,41 @@ def merge_semesters_count(data)
     end
 
     def map_enrollments(enrollment_term, term_id)
-      enrollment_term.map do |course|
-        next unless course[:role] == 'Student'
-        next if law_student? && !academic_careers.include?(course[:academicCareer])
-        mapped_course = course_info course
+      enrollment_term.map do |enrollment|
+        next unless enrollment[:role] == 'Student'
+        next if exclude_enrollment?(enrollment)
+        mapped_enrollment = course_info enrollment
         if @filtered
-          mapped_course.delete :url
+          mapped_enrollment.delete :url
         else
-          process_unfiltered_course(course, term_id)
+          process_unfiltered_enrollment(enrollment, term_id)
+        end
+        mapped_enrollment
+      end.compact
+    end
+
+    def exclude_enrollment?(enrollment)
+      if law_student?
+        if current_academic_roles['lawJointDegree']
+          if !['GRAD','LAW'].include?(enrollment[:academicCareer])
+            return true
+          end
+        else
+          if !academic_careers.include?(enrollment[:academicCareer])
+            return true
+          end
         end
-        mapped_course
       end
+      false
+    end
+
+    def current_academic_roles
+      @current_academic_roles ||= MyAcademics::MyAcademicRoles.new(@uid).get_feed[:current]
     end
 
-    def process_unfiltered_course(course, term_id)
+    def process_unfiltered_enrollment(enrollment, term_id)
```

Comment:
This seems like a good place to inject an `Enrollment` object, even if it's just defined privately within this class

Reply:
@developish - Yeah that would be nice if I had more time to deliver this.